hathor/merged_mining/digibyte_rpc.py

```diff
@@ -0,0 +1,151 @@
+from abc import ABC, abstractmethod
+from itertools import count
+from typing import Any, Callable, Dict, Iterator, List, Optional, Union, cast
+
+from aiohttp import BasicAuth, ClientSession
+from structlog import get_logger
+
+logger = get_logger()
+
+
+class RPCFailure(Exception):
```

Comment:
You already have one of this in the `bitcoin_rpc.py`, can we use the same?

Reply:
I guess I could. I can make an `rpc_commons` module for this.

Reply:
Done, made a `jsonrpc` module instead.